controllers/componentdetectionquery_controller.go

```diff
@@ -129,67 +165,39 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 					return ctrl.Result{}, nil
 				}
 				log.Info(fmt.Sprintf("cloned from %s to path %s... %v", source.URL, clonePath, req.NamespacedName))
-
-				devfilesMap, devfilesURLMap, dockerfileContextMap, err = devfile.ScanRepo(log, r.AlizerClient, clonePath, maxDevfileDiscoveryDepth, r.DevfileRegistryURL)
-				if err != nil {
-					if _, ok := err.(*devfile.NoDevfileFound); !ok {
-						log.Error(err, fmt.Sprintf("Unable to find devfile(s) in repo %s due to an error %s, exiting reconcile loop %v", source.URL, err.Error(), req.NamespacedName))
-						r.SetCompleteConditionAndUpdateCR(ctx, &componentDetectionQuery, err)
-						return ctrl.Result{}, nil
-					}
-				}
-			} else {
-				log.Info(fmt.Sprintf("Since this is not a multi-component, attempt will be made to read devfile at the root dir... %v", req.NamespacedName))
-
-				isDockerfilePresent := false
-				isDevfilePresent := false
-
-				var gitURL string
-				if gitToken == "" {
-					gitURL, err = util.ConvertGitHubURL(source.URL, source.Revision)
+				if !isDevfilePresent {
```

Comment:
Dont you think this should be at line 151? https://github.com/redhat-appstudio/application-service/pull/132/files#diff-ff25076a2825352e686733573311ad782c88ad07b11ec3cadb65bb54e011b438R151

Because if a dockerfile is not present in line 151, the rest of the if block does nothing for it. Seems like extra nesting.. if a devfile is not found in line 151 - clone and detect components using alizer to see if its multi component

Reply:
as explained in https://github.com/redhat-appstudio/application-service/pull/132#discussion_r900433116
This is simplifies https://github.com/redhat-appstudio/application-service/blob/9c6a8ee09b046e66e90957dab8be574925ee3bb2/controllers/componentdetectionquery_controller.go#L177-L192